src/edid.h

```diff
@@ -22,15 +22,150 @@
 
 /* Author: Soren Sandmann <sandmann@redhat.com> */
 
+enum  interface {
+	UNDEFINED,
+	DVI,
+	HDMI_A,
+	HDMI_B,
+	MDDI,
+	DISPLAY_PORT
+};
+
+enum color_type {
+	UNDEFINED_COLOR,
+	MONOCHROME,
+	RGB,
+	OTHER_COLOR
+};
+
+enum stereo_type {
+	NO_STEREO,
+	FIELD_RIGHT,
+	FIELD_LEFT,
+	TWO_WAY_RIGHT_ON_EVEN,
+	TWO_WAY_LEFT_ON_EVEN,
+	FOUR_WAY_INTERLEAVED,
+	SIDE_BY_SIDE
+};
+
+struct timing {
+	int width;
+	int height;
+	int frequency;
+};
+
+struct detailed_timing {
+	int pixel_clock;
+	int h_addr;
+	int h_blank;
+	int h_sync;
+	int h_front_porch;
+	int v_addr;
+	int v_blank;
+	int v_sync;
+	int v_front_porch;
+	int width_mm;
+	int height_mm;
+	int right_border;
+	int top_border;
+	int interlaced;
+	enum stereo_type stereo;
+
+	int digital_sync;
+	union {
+		struct {
+			int bipolar;
+			int serrations;
+			int sync_on_green;
+		} analog;
+
+		struct {
+			int composite;
+			int serrations;
+			int negative_vsync;
+			int negative_hsync;
+		} digital;
+	};
+};
+
 struct monitor_info {
 	int checksum;
+	char manufacturer_code[4];
+	int product_code;
+	unsigned int serial_number;
+
+	int production_week;	/* -1 if not specified */
+	int production_year;	/* -1 if not specified */
+	int model_year;		/* -1 if not specified */
+
+	int major_version;
+	int minor_version;
+
+	int is_digital;
+
+	union {
+		struct {
+			int bits_per_primary;
+			enum interface interface;
+			int rgb444;
+			int ycrcb444;
+			int ycrcb422;
+		} digital;
+
+		struct {
+			double video_signal_level;
+			double sync_signal_level;
+			double total_signal_level;
+
+			int blank_to_black;
+
+			int separate_hv_sync;
+			int composite_sync_on_h;
+			int composite_sync_on_green;
+			int serration_on_vsync;
+			enum color_type color_type;
+		} analog;
+	};
+
+	int width_mm;		/* -1 if not specified */
+	int height_mm;		/* -1 if not specified */
+	double aspect_ratio;	/* -1.0 if not specififed */
+
+	double gamma;		/* -1.0 if not specified */
+
+	int standby;
+	int suspend;
+	int active_off;
+
+	int srgb_is_standard;
+	int preferred_timing_includes_native;
+	int continuous_frequency;
+
+	double red_x;
+	double red_y;
+	double green_x;
+	double green_y;
+	double blue_x;
+	double blue_y;
+	double white_x;
+	double white_y;
+
+	struct timing established[24];	/* Terminated by 0x0x0 */
+	struct timing standard[8];
+
+	int n_detailed_timings;
+	struct detailed_timing detailed_timings[4];	/* If monitor has a preferred
```

Comment:
Here the right-hand side comment could be better placed, before n_detailed_timings, to document this block.  Also, for multi-line comments, please use a leading empty row:

```C
/*
 * like this
 * it makes it 
 * more like 
 * readable
 * and stuff
 */
```